pkg/lvm/lvm_util.go

```diff
@@ -265,3 +271,82 @@ func DestroySnapshot(snap *apis.LVMSnapshot) error {
 func getLVMSnapName(snapName string) string {
 	return strings.TrimPrefix(snapName, "snapshot-")
 }
+
+func decodeVgsJSON(raw []byte) ([]apis.VolumeGroup, error) {
+	output := &struct {
+		Report []struct {
+			VolumeGroups []map[string]string `json:"vg"`
```

Comment:
If we specify all the json keys correctly, cant we directly unmarshal to `apis.VolumeGroup` ? or some other struct instead of using `[]map[string]string`. It is confusing.

Reply:
Issue is lvm json output always returns json objects values in string regardless if it's numeric(e.g vg size) or boolean which is unfortunate. See the example below. So, it may not a good idea to mask the typing(int/bool/string) or size SI units information with fields under `apis.VolumeGroup` just for the benefit of easy parsing since in that case one will end up parsing these string fields everytime as & when we'll be using/consuming it (e.g in GetCapacity we need to compare vg size).

```
{
  "report": [
    {
      "vg": [
        {
          "vg_fmt": "lvm2",
          "vg_uuid": "VPPIu3-FCFI-svbR-lljg-aqID-ryuN-24ShtQ",
          "vg_name": "lvmvg",
          "vg_attr": "wz--n-",
          "vg_permissions": "writeable",
          "vg_extendable": "extendable",
          "vg_size": "32.00g",
          "vg_free": "32.00g",
          "vg_extent_size": "4.00m",
          "vg_extent_count": "8191",
          "vg_free_count": "8191",
          "pv_count": "1",
          "lv_count": "0",
          "snap_count": "0",
          "vg_seqno": "23"
        }
      ]
    }
  ]
}
```